apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentDAO.java

```diff
@@ -526,4 +535,28 @@ public Flux<WorkspaceAndResourceId> getExperimentWorkspaces(@NonNull Set<UUID> e
                         row.get("workspace_id", String.class),
                         row.get("id", UUID.class))));
     }
+
+    public Mono<Long> delete(UUID id) {
+        Preconditions.checkArgument(Objects.nonNull(id), "Argument 'id' must not be null");
```

Comment:
Minor: just use Lombok `@NonNull`, it's more concise.

Reply:
I kept it because we changed to a Set of UUIDs, ok?

Reply:
Correct. After changing to a set, the checkArgument without NonNull annotation is preferable.